packages/react-dom/src/__tests__/DOMPropertyOperations-test.js

```diff
@@ -381,6 +381,164 @@ describe('DOMPropertyOperations', () => {
       expect(eventHandler).toHaveBeenCalledTimes(2);
     });
 
+    it('<input is=...> should have the same onChange/onInput/onClick behavior as <input>', () => {
+      const container = document.createElement('div');
+      document.body.appendChild(container);
+      const regularOnInputHandler = jest.fn();
+      const regularOnChangeHandler = jest.fn();
+      const regularOnClickHandler = jest.fn();
+      const customOnInputHandler = jest.fn();
+      const customOnChangeHandler = jest.fn();
+      const customOnClickHandler = jest.fn();
+      ReactDOM.render(
+        <div>
+          <input
+            onInput={regularOnInputHandler}
+            onChange={regularOnChangeHandler}
+            onClick={regularOnClickHandler}
+          />
+          <input
+            is="my-custom-element"
+            onInput={customOnInputHandler}
+            onChange={customOnChangeHandler}
+            onClick={customOnClickHandler}
+          />
+        </div>,
+        container,
+      );
+
+      const regularInput = container.querySelector(
+        'input:not([is=my-custom-element])',
+      );
+      const customInput = container.querySelector(
+        'input[is=my-custom-element]',
+      );
+      expect(regularInput).not.toBe(customInput);
+
+      regularInput.dispatchEvent(new Event('input', {bubbles: true}));
+      regularInput.dispatchEvent(new Event('change', {bubbles: true}));
+      regularInput.dispatchEvent(new Event('click', {bubbles: true}));
+      customInput.dispatchEvent(new Event('input', {bubbles: true}));
+      customInput.dispatchEvent(new Event('change', {bubbles: true}));
+      customInput.dispatchEvent(new Event('click', {bubbles: true}));
+
+      expect(customOnInputHandler).toHaveBeenCalledTimes(
+        regularOnInputHandler.mock.calls.length,
```

Comment:
This is kind of fragile. (What if we change some heuristic and all of these are 0.) Can you make the test a bit more specific? Also ideally you'd verify that the handler runs as early as possible, and not at the end of the function. So that if there's a timing difference, it is visible.

Reply:
I added expectations after each dispatchEvent and put specific numbers in